### PR TITLE
feat: use rangefunc for iterators

### PR DESCRIPTION
**Pull request by utkuozdemir**
Give rangefunc a try.

### CODE REVIEW

**Comment by DmitriyMV** [comment]
Depends on https://github.com/dominikh/go-tools/issues/1494

File: pkg/safe/state.go [hunk 1]
```diff
@@ -283,10 +284,27 @@ func (l *List[T]) Find(fn func(T) bool) (T, bool) {
...
+
+			if !yield(res) {
+				return
+			}
+		}
+	}
+}
+
 // ListIterator is a generic iterator over resource.Resource slice.
 type ListIterator[T any] struct {
```
Comment (DmitriyMV) [inline pkg/safe/state.go:309]: Needs deprecation notice.

File: pkg/safe/state.go [hunk 2]
```diff
@@ -283,10 +284,27 @@ func (l *List[T]) Find(fn func(T) bool) (T, bool) {
...
 
 // Iterator returns a new iterator over the list.
+//
+// Deprecated: use [List.Iter] instead.
 func (l *List[T]) Iterator() ListIterator[T] {
 	return ListIterator[T]{pos: 0, list: *l}
 }
 
+// Iter returns a new rangefunc iterator over the list.
+func (l *List[T]) Iter() iter.Seq[T] {
```
Comment (DmitriyMV) [inline pkg/safe/state.go:294]: All because if this https://pkg.go.dev/iter#hdr-Naming_Conventions

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/cosi-project/runtime
 
-go 1.22.0
+go 1.22.1
```
Comment (DmitriyMV) [inline go.mod:3]:
```suggestion
go 1.23.0
```

File: pkg/safe/state.go (hunk 1 above)
Comment (utkuozdemir) [reply pkg/safe/state.go:309]: can do, but also feel free to take this over if you want to push this faster forward.

Comment (DmitriyMV) [reply pkg/safe/state.go:309]: Okay, I'll take it.

Comment (utkuozdemir) [reply pkg/safe/state.go:309]: 🙏

**Review by DmitriyMV — APPROVED** [review]

**Pull request merged**